evalml/tests/automl_tests/test_automl.py

```diff
@@ -5363,3 +5363,83 @@ def test_automl_respects_iterative_pipeline_order(X_y_binary, AutoMLTestEnv):
         searched_model_families,
         key=lambda model_family: _ESTIMATOR_FAMILY_ORDER.index(model_family),
     )
+
+
+def test_get_ensembler_input_pipelines(X_y_binary, AutoMLTestEnv):
+    X, y = X_y_binary
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_iterations=_get_first_stacked_classifier_no(),
+        objective="Log Loss Binary",
+        ensembling=True,
+    )
+
+    score_side_effect = [
+        {"Log Loss Binary": score}
+        for score in np.arange(
+            0, -1 * automl.max_iterations * automl.data_splitter.get_n_splits(), -0.1
+        )
+    ]  # Decreases with each call
+
+    test_env = AutoMLTestEnv("binary")
+    with test_env.test_context(mock_score_side_effect=score_side_effect):
+        automl.search()
+
+    best_pipeline_ids = [
+        pipeline["id"]
+        for pipeline in list(automl._automl_algorithm._best_pipeline_info.values())
+    ]
+    best_pipeline_ids.sort()
```

Comment:
Q: What do you think about just sorting the output for `get_ensembler_input_pipelines`? Is there ever a case where we wouldn't want to?

Reply:
@angela97lin I think the pipeline IDs come out in model family order. In the case of the IterativeAlgorithm, it comes out in the order of the first batch. It's a minor detail but this is the same order that is fed into `_create_ensemble`, so all of the input pipelines in the ensemble pipeline's graph are in this order from top-down.

Reply:
@christopherbunn Isn't `_best_pipeline_info` a dictionary? So the order isn't guaranteed, unless we're assuming python order from dict --> list is the order in which items are added to the dict?

Reply:
@angela97lin From Python 3.7 onwards it's guaranteed to be insertion order: https://stackoverflow.com/questions/60848/how-do-you-retrieve-items-from-a-dictionary-in-the-order-that-theyre-inserted

Reply:
Ah right! 😂 
Okay back to the original question then: is it better to sort or not? Sounds like there's no benefit to doing so LGTM! My original question was because I noticed we were sorting the outputs in the tests to compare and wondered if there was any value to changing the method to sort. It's not too difficult to add later if we change our minds :)

Reply:
Ah gotcha! Yeah to be clear I think it would be best to retain insertion order for the output rather than sort it, but it's not necessary for now. I have it sorted for this test case so it's easier to compare values.